src/ucp/core/ucp_am.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (C) Los Alamos National Security, LLC. 2019 ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "ucp_ep.h"
+
+#define AM_BLOCK 16
+
+
+typedef struct {
+    uint32_t     length;     /* length of an AM. Ideally it would be size_t
+                              * but we want to keep this struct at 64 bits
+                              * to fit in uct_ep_am_short header. MAX_SHORT
+                              * or b/zcopy MTU
+                              * should be much smaller than this anyway */
+    uint16_t     am_id;      /* Index into callback array */
+    uint16_t     flags;      /* currently unused in this header 
+                                because replies require long header
+                                defined by @ref ucp_am_send_flags */
+} UCS_S_PACKED ucp_am_hdr_t;
+
+typedef struct {
+    ucp_am_hdr_t super;
+    uintptr_t    ep_ptr;
+} UCS_S_PACKED ucp_am_reply_hdr_t;
+
+typedef struct {
+    size_t            total_size; /* length of buffer needed for all data */
+    uint64_t          msg_id;     /* method to match parts of the same AM */
+    uintptr_t         ep;         /* end point ptr, used for maintaing list 
+                                     of arrivals */
+    size_t            offset;     /* how far this message goes into large
+                                     the entire AM buffer */
+    uintptr_t         ep_ptr;     /* pointer for sending ack */
+    uint16_t          am_id;      /* index into callback array */
+    uint16_t          flags;      /* flags that indicate how to handle message
```

Comment:
i couldn't find where ucp_am_send_flags  is defined?

Reply:
should be ucp_send_am_flags, will fix it